scripts/tools/metric.yml

```diff
@@ -185,25 +186,29 @@ params:
     - type: uint32_t*
       name: pSetCount
       desc: |
-            [in,out] pointer to number of metric sets.
+            [in,out][optional] pointer to number of metric sets.
             if *pSetCount is zero, then the driver shall update the value with the total number of metric sets to be calculated.
             if *pSetCount is greater than the number available in the raw data buffer, then the driver shall update the value with the actual number of metric sets to be calculated.
-            if *pSetCount is is less than the number available in the raw data buffer, then the driver shall calculate the metrics for up to that number of sub-devices.
+            if *pSetCount is less than the number of sets available, then driver shall only calculate that number of sets.
     - type: uint32_t*
-      name: pTotalMetricValueCount
+      name: pMetricValueCount
       desc: |
-            [in,out] pointer to number of the total number of metric values calculated, for all metric sets.
-            if *pSetCount is zero, then the driver shall update the value with the total number of metric values to be calculated.
-            if *pSetCount is greater than the number available in the raw data buffer, then the driver shall update the value with the actual number of metric values to be calculated.
+            [in,out][optional] pointer to the total number of metric values calculated, it is the sum of all elements in pMetricCountPerSet array.
+            if *pMetricValueCount is zero, then the driver shall update the value with the total number of metric values to be calculated.
+            if *pMetricValueCount is greater than the number available in the raw data buffer, then the driver shall update the value with the actual number of metric values to be calculated.
     - type: uint32_t*
-      name: pMetricCounts
+      name: pMetricCountPerSet
       desc: |
             [in,out][optional][range(0, *pSetCount)] buffer of metric counts per metric set.
+            Driver shall update the value with the number of metric values calculated for that set.
+            If count for a given set is greater than the number of metric values available in the raw data buffer in that set, then the driver
+            shall update the value with the actual number of metric values to be calculated for that set.
```

Comment:
`then the driver shall update the value with the actual number of metric values calculated for that set.` 

remove "to be"